### PR TITLE
Add log metadata

### PR DESCRIPTION
**Pull request by EGiataganas**
We added log metadata method to ensure that we'll be able to see
the visit_id throughout the Booking Request Creator process.

### CODE REVIEW

File: spec/services/booking_request_creator_spec.rb [hunk 1]
```diff
@@ -54,13 +54,13 @@
         slot_option_1: '2015-01-02T09:00/10:00',
         slot_option_2: '2015-01-03T09:00/10:00',
         slot_option_3: '2015-01-04T09:00/10:00'
-      )
+      ).and_return double(Visit, id: 2)
```
Comment (tatyree) [inline spec/services/booking_request_creator_spec.rb]: I'd like to see a spec that explains why `id` is necessary, please.

Comment (threedaymonk) [inline spec/services/booking_request_creator_spec.rb]: Something like:

```ruby
expect(LogStasher.request_context).to have_attribute(id: 1234)
```

perhaps.

File: app/services/booking_request_creator.rb [hunk 2]
```diff
@@ -3,11 +3,16 @@ def create!(prisoner_step, visitors_step, slots_step)
...
       PrisonMailer.request_received(visit).deliver_later
+      add_log_metadata visit_id: visit.id
     }
   end
 
 private
 
+  def add_log_metadata(hash)
+    LogStasher.request_context.merge!(hash)
+  end
```
Comment (tatyree) [inline app/services/booking_request_creator.rb]: I'd like to see a couple of unit tests in the spec to give this and the `id` parameters passed by the doubles a bit of context, please.  If I didn't know the history, I would not find it immediately clear why we were logging something that would potentially be in the database.

File: spec/services/booking_request_creator_spec.rb (hunk 1 above)
Comment (tatyree) [inline spec/services/booking_request_creator_spec.rb]: @threedaymonk :+1:

Comment (EGiataganas) [inline spec/services/booking_request_creator_spec.rb]: :+1:

File: spec/services/booking_request_creator_spec.rb [hunk 3]
```diff
@@ -54,13 +54,13 @@
...
 
     described_class.new.create! prisoner_step, visitors_step, slots_step
   end
 
   it 'emails the prison' do
-    visit = double(Visit)
+    visit = double(Visit, id: 2)
     allow(Visit).to receive(:create!).and_return(visit)
```
Comment (tatyree) [inline spec/services/booking_request_creator_spec.rb]: Could you DRY this and the visit declaration out, please? I'd suggest something like: 

```
context 'emailing and logging' do 
  let(:visit) { double(Visit, id: 2) }
  before do
    allow(Visit).to receive(:create!).and_return(visit)
  end

  it 'emails the prison' do 
    expect(PrisonMailer).to receive(:request_received).with(visit)...
   ...

 it 'logs the visit id' do 
   expect(LogStasher.request_context).to match(visit_id: 2) 
```

Comment (EGiataganas) [inline spec/services/booking_request_creator_spec.rb]: @tatyree Sounds great. I will also do something like

```
    after do
      described_class.new.create! prisoner_step, visitors_step, slots_step
    end
```

File: spec/services/booking_request_creator_spec.rb (hunk 1 above)
Comment (dan-palmer) [inline spec/services/booking_request_creator_spec.rb]: Should be using `instance_double` here.

File: spec/services/booking_request_creator_spec.rb [hunk 4]
```diff
@@ -33,6 +33,8 @@
     double(Mail::Message, deliver_later: nil)
   }
 
+  subject { described_class.new }
```
Comment (threedaymonk) [inline spec/services/booking_request_creator_spec.rb]: I think this is implicit.

**Pull request merged**